src/libraries/System.Private.CoreLib/src/System/SearchValues/IndexOfAnyAsciiSearcher.cs

```diff
@@ -1017,73 +1100,682 @@ public static int LastIndexOfAny<TNegator>(ref byte searchSpace, int searchSpace
+                    if (result != Vector128<byte>.Zero)
+                    {
+                        return ComputeLastIndex<short, TNegator>(ref searchSpace, ref currentSearchSpace, result);
+                    }
+                }
+                while (Unsafe.IsAddressGreaterThan(ref currentSearchSpace, ref twoVectorsAfterStart));
+            }
+
+            // We have 1-16 characters remaining. Process the first and last vector in the search space.
+            // They may overlap, but we'll handle that in the index calculation if we do get a match.
+            Debug.Assert(searchSpaceLength >= Vector128<short>.Count, "We expect that the input is long enough for us to load a whole vector.");
+            {
+                ref short oneVectorAfterStart = ref Unsafe.Add(ref searchSpace, Vector128<short>.Count);
+
+                ref short secondVector = ref Unsafe.IsAddressGreaterThan(ref currentSearchSpace, ref oneVectorAfterStart)
+                    ? ref Unsafe.Subtract(ref currentSearchSpace, Vector128<short>.Count)
+                    : ref searchSpace;
+
+                Vector128<short> source0 = Vector128.LoadUnsafe(ref searchSpace);
+                Vector128<short> source1 = Vector128.LoadUnsafe(ref secondVector);
+
+                Vector128<byte> result = IndexOfAnyLookupUniqueLowNibble<TNegator>(source0, source1, values);
+                if (result != Vector128<byte>.Zero)
+                {
+                    return ComputeLastIndexOverlapped<short, TNegator>(ref searchSpace, ref secondVector, result);
+                }
+            }
+
+            return -1;
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        [CompExactlyDependsOn(typeof(Ssse3))]
+        [CompExactlyDependsOn(typeof(AdvSimd))]
+        [CompExactlyDependsOn(typeof(PackedSimd))]
+        public static bool ContainsAny<TNegator>(ref byte searchSpace, int searchSpaceLength, ref UniqueLowNibbleState state)
+            where TNegator : struct, INegator =>
+            IndexOfAnyCore<bool, TNegator, ContainsAnyResultMapper<byte>>(ref searchSpace, searchSpaceLength, ref state);
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        [CompExactlyDependsOn(typeof(Ssse3))]
+        [CompExactlyDependsOn(typeof(AdvSimd))]
+        [CompExactlyDependsOn(typeof(PackedSimd))]
+        public static int IndexOfAny<TNegator>(ref byte searchSpace, int searchSpaceLength, ref UniqueLowNibbleState state)
+            where TNegator : struct, INegator =>
+            IndexOfAnyCore<int, TNegator, IndexOfAnyResultMapper<byte>>(ref searchSpace, searchSpaceLength, ref state);
+
+        [CompExactlyDependsOn(typeof(Ssse3))]
+        [CompExactlyDependsOn(typeof(AdvSimd))]
+        [CompExactlyDependsOn(typeof(PackedSimd))]
+        private static TResult IndexOfAnyCore<TResult, TNegator, TResultMapper>(ref byte searchSpace, int searchSpaceLength, ref UniqueLowNibbleState state)
```

Comment:
The structure of these implementations looks very similar to the structure of other ones. Is there any way to share more of these routines without sacrificing throughput? More generics?

Reply:
Yeah it really is a copy-paste of the ASCII one, swapping out the `IndexOfAnyLookup` implementation.

The state needed for this one happens to look the same as that for ASCII (`BitVector256 Lookup + Vector128<byte> Bitmap`), even if the contents differ.
It should be possible to de-dup the two with an extra `IRuntimeConst` generic.